app/services/flow/flow_state_machine.rb

```diff
@@ -21,9 +21,13 @@ def show
     def update
       step = current_step
       result = flow.handle(step)
+
       if @analytics_id
         increment_step_name_counts
-        analytics.track_event(analytics_submitted, result.to_h.merge(analytics_properties))
+        analytics.public_send(
+          flow.step_handler(step).analytics_submitted_event,
+          **result.to_h.merge(analytics_properties),
+        )
```

Comment:
Trying to think how we validate that these methods are defined. I suppose we can rely on the fact that an error would be thrown in feature specs if they weren't?

Reply:
Another thing I discovered, the guards for `if @analytics_id` basically prevented us from trying to log information, I am trying removing that here to see we discover more errors: https://github.com/18F/identity-idp/pull/7213